apps/token/src/routes/governance/components/proposal-detail-header/proposal-header.tsx

```diff
@@ -0,0 +1,104 @@
+import { useTranslation } from 'react-i18next';
+import { Lozenge } from '@vegaprotocol/ui-toolkit';
+import type { ReactNode } from 'react';
+import type { Proposals_proposals } from '../../proposals/__generated__/Proposals';
+
+export const ProposalHeader = ({
+  proposal,
+}: {
+  proposal: Proposals_proposals;
+}) => {
+  const { t } = useTranslation();
+  const { change } = proposal.terms;
+
+  let headerText: string;
+  let detailsOne: ReactNode;
+  let detailsTwo: ReactNode;
+
+  switch (change.__typename) {
```

Comment:
chore (non-blocking): i'd extract this to a function above the component, tbh, to make it a bit cleaner - you wouldn't need to attach types to variables which are undefined at the time of creation + giving an extra benefit to easily hook it into a memo function:

```
const getChildren = (change) => {
  switch (..) {
    case : return {
      headerText: '...',
      detailOne: <>...</>
      detailTwo: <>...</>
    }
  }
}

export const ProposalHeader = ({ proposal }) => {
  const { change } = proposal.terms;
  
  const { header, detailOne, detailTwo } = useMemo(() => getChildren(change), [change]);
  ...
```